include/openthread/thread_ftd.h

```diff
@@ -953,6 +953,15 @@ void otThreadGetNextHopAndPathCost(otInstance *aInstance,
                                    uint16_t   *aNextHopRloc16,
                                    uint8_t    *aPathCost);
 
+/**
+ * Calculates and updates value of CSL Frame Request Ahead, based on bus speed, bus latency and
+ * `OPENTHREAD_CONFIG_MAC_CSL_REQUEST_AHEAD_US`.
+ *
+ * @param[in] aInstance A pointer to an OpenThread instance.
+ *
+ */
+void otThreadUpdateFrameRequestAhead(otInstance *aInstance);
```

Comment:
Adding this as a new public OT API is a viable option, and I see that it's called from the radio platform implementation whenever the bus latency value changes. 

However, we could also consider making this an `otPlatRadio` callback instead, something like `otPlatRadioHandleBusLatencyChanged()`. This callback would then trigger the necessary updates to CSL-related calculations.

This approach would make it clear that only the radio platform can signal this change, not a user of the OpenThread stack. Thoughts?

Reply:
Yeah I was wondering if that can be done a few days back. I encountered the issue with calling `UpdateFrameRequestAhead` from `posix/radio.cpp` as I couldn't use `locator_getters.hpp`  for `AsCoreType(aInstance).Get<CslTxScheduler>()` in posix. I'm not sure if there is some other way to call csltxscheduler's method instead of using the API I created?

Reply:
Added the callback, thanks for the hint @Irving-cl !